R/database-snomed.R

```diff
@@ -0,0 +1,329 @@
+#'   - `Snapshot/Refset`
+#'
+#' @details
+#' The function reads all `.txt` files within the Terminology and Refset snapshot folders
+#' using `data.table::fread()`, with all columns imported as character type to preserve data fidelity.
+#' It constructs:
+#' - A joined table of SNOMED CT concept and description data.
+#' - The relationship table of concept relationships.
+#' - The ICD-10 map subset from the Extended Map refset.
+#'
+#' The function automatically normalizes file names by removing suffixes like `_GB_<date>.txt`.
+#'
+#' @return
+#' A named list with three elements:
+#' \describe{
+#'   \item{`sct_description`}{A `data.table` combining SNOMED CT concepts and descriptions.}
+#'   \item{`sct_relationship`}{A `data.table` containing concept relationships.}
+#'   \item{`sct_icd10`}{A `data.table` containing SNOMED CT–to–ICD-10 map entries.}
+#' }
+#'
+#' @examples
+#' \dontrun{
+#' # Read a local SNOMED CT UK Monolith Edition
+#' snomed <- read_snomed_ct_uk_monolith("~/SNOMEDCT_Release_UK")
+#'
+#' # View available datasets
+#' names(snomed)
+#'
+#' # Inspect a few rows of each
+#' head(snomed$sct_description)
+#' head(snomed$sct_relationship)
+#' head(snomed$sct_icd10)
+#' }
+#'
+#' @export
+read_snomed_ct_uk_monolith <- function(snomed_ct_uk_monolith_dir) {
+  if (!dir.exists(snomed_ct_uk_monolith_dir)) {
+    cli::cli_abort(
+      "Directory does not exist: {.path {snomed_ct_uk_monolith_dir}}"
+    )
+  }
+
+  snomed_monolith_terminology <-
+    file.path(snomed_ct_uk_monolith_dir, "Snapshot", "Terminology") |>
+    list.files(full.names = TRUE) |>
+    purrr::set_names(nm = fs::path_file) |>
+    purrr::map(
+      ~ data.table::fread(
+        .x,
+        sep = "\t",
+        colClasses = "character",
+        quote = ""
+      )
+    )
+
+  snomed_monolith_refset <- file.path(
+    snomed_ct_uk_monolith_dir,
+    "Snapshot",
+    "Refset"
+  ) |>
+    list.files(full.names = TRUE) |>
+    purrr::set_names(nm = fs::path_file) |>
+    purrr::map(
+      ~ .x |>
+        list.files(full.names = TRUE) |>
+        purrr::set_names(nm = fs::path_file) |>
+        purrr::map(
+          ~ data.table::fread(
+            .x,
+            sep = "\t",
+            colClasses = "character",
+            quote = ""
+          )
+        )
+    )
+
+  names(snomed_monolith_terminology) <-
+    stringr::str_replace(
+      string = names(snomed_monolith_terminology),
+      pattern = "_GB_[0-9]+\\.txt",
+      ".txt"
+    )
+
+  valid_slots <- intersect(
+    c("Content", "Language", "Map", "Metadata"),
+    names(snomed_monolith_refset)
+  )
+  for (slot in valid_slots) {
+    if (!is.null(snomed_monolith_refset[[slot]])) {
+      names(snomed_monolith_refset[[slot]]) <-
+        stringr::str_replace(
+          names(snomed_monolith_refset[[slot]]),
+          "_GB_[0-9]+\\.txt",
+          ".txt"
+        )
+    }
+  }
+
+  sct_description <- snomed_monolith_terminology$`sct2_Description_MONOSnapshot-en.txt` |>
```

Comment:
I'd just use `[[` here to avoid having to use the backticks

```suggestion
  sct_description <- snomed_monolith_terminology[["sct2_Description_MONOSnapshot-en.txt"]] |>
```